### ISSUE TITLE
How to get the label of coco128-seg

### ISSUE THREAD

**Issue by small-code-cat** [ultralytics/yolov5#9816]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

16 0.597625 0.561987 0.516 0.481025 0.435891 0.430439 0.357281 0.3009 0.334609 0.294833 0.35275 0.280669 0.402641 0.25841 0.432859 0.17341 0.512969 0.100544 0.677734 0.0863808 0.785047 0.167343 0.87725 0.351506 0.856094 0.452699 0.878766 0.515439 0.847016 0.596381 0.797141 0.596381 0.730641 0.466862 0.655062 0.529603 0.615766 0.553891

As for the above data, I know that the first number represents category. What do other numbers mean? How can I get this annotation file from the json file of labelme?

**Comment by glenn-jocher** [comment]
class followed by xy, xy... coordinates. If you have COCO segmentation format labels you can export them to YOLO format with https://github.com/ultralytics/JSON2YOLO

**Comment by small-code-cat** [comment]
> class followed by xy, xy... coordinates. If you have COCO segmentation format labels you can export them to YOLO format with https://github.com/ultralytics/JSON2YOLO

I want to ask what are the calculation formulas of x and y？

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about YOLOv5 🚀 **dataset formatting**. The xy coordinates for segmentation are the same normalized coordinates used for box centers.

To train correctly your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths) and 2) a class `names` dictionary:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes (80 COCO classes)
names:
  0: person
  1: bicycle
  2: car
  ...
  77: teddy bear
  78: hair drier
  79: toothbrush

```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<p align="center"><img width="750" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg"></p>

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>

Good luck 🍀 and let us know if you have any other questions!

**Comment by small-code-cat** [comment]
> 

Thank you very much. This is effective

**Comment by small-code-cat** [comment]
> ### Search before asking
> * [x]  I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.
> 
> ### Question
> 16 0.597625 0.561987 0.516 0.481025 0.435891 0.430439 0.357281 0.3009 0.334609 0.294833 0.35275 0.280669 0.402641 0.25841 0.432859 0.17341 0.512969 0.100544 0.677734 0.0863808 0.785047 0.167343 0.87725 0.351506 0.856094 0.452699 0.878766 0.515439 0.847016 0.596381 0.797141 0.596381 0.730641 0.466862 0.655062 0.529603 0.615766 0.553891
> 
> As for the above data, I know that the first number represents category. What do other numbers mean? How can I get this annotation file from the json file of labelme?
> 
> ### Additional
> _No response_

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/44658572/196035712-ca18e924-448f-4db6-9db4-33aa6296f9a6.png">
As shown in the figure above, I use the trained segmentation model to identify this picture of coco in the code. How can I get normalized data similar to those in label txt?

**Comment by small-code-cat** [comment]
Sorry, can you understand the question I asked above？

**Comment by gjr2019** [comment]
> Sorry, can you understand the question I asked above？

I has the same problem like yours, did you get it?

**Comment by gjr2019** [comment]
> class followed by xy, xy... coordinates. If you have COCO segmentation format labels you can export them to YOLO format with https://github.com/ultralytics/JSON2YOLO

hi, there is a question that how to convert labelme segmentation json file to yolo txt file , JSON2YOLO seems can't convert the labelme segmentation json file to yolo txt file successfully

**Comment by small-code-cat** [comment]
> hi, there is a question that how to convert labelme segmentation json file to yolo txt file , JSON2YOLO seems can't convert the labelme segmentation json file to yolo txt file successfully

Yes, you need to write your own script to convert

**Comment by gjr2019** [comment]
> > hi, there is a question that how to convert labelme segmentation json file to yolo txt file , JSON2YOLO seems can't convert the labelme segmentation json file to yolo txt file successfully
> 
> Yes, you need to write your own script to convert

thanks, and i have another question as follows:
16 0.597625 0.561987 0.516 0.481025 0.435891 0.430439 0.357281 0.3009 0.334609 0.294833 0.35275 0.280669 0.402641 0.25841 0.432859 0.17341 0.512969 0.100544 0.677734 0.0863808 0.785047 0.167343 0.87725 0.351506 0.856094 0.452699 0.878766 0.515439 0.847016 0.596381 0.797141 0.596381 0.730641 0.466862 0.655062 0.529603 0.615766 0.553891 
that's your previous question.
What I want to know is if these coordinates are in any order? and how to nomalize these coordinates to 0-1.

**Comment by small-code-cat** [comment]
> > > hi, there is a question that how to convert labelme segmentation json file to yolo txt file , JSON2YOLO seems can't convert the labelme segmentation json file to yolo txt file successfully
> > 
> > 
> > Yes, you need to write your own script to convert
> 
> thanks, and i have another question as follows: 16 0.597625 0.561987 0.516 0.481025 0.435891 0.430439 0.357281 0.3009 0.334609 0.294833 0.35275 0.280669 0.402641 0.25841 0.432859 0.17341 0.512969 0.100544 0.677734 0.0863808 0.785047 0.167343 0.87725 0.351506 0.856094 0.452699 0.878766 0.515439 0.847016 0.596381 0.797141 0.596381 0.730641 0.466862 0.655062 0.529603 0.615766 0.553891 that's your previous question. What I want to know is if these coordinates are in any order? and how to nomalize these coordinates to 0-1.

You can see that the answers given by the author are very detailed

**Comment by WongVi** [comment]
@small-code-cat 
Could you please share the python file to convert labelme format data into yolo segmentation format training data?

**Comment by muhemuhe427** [comment]
I've wrote a script to transfer the label .json as .txt , and It is proved feasible for training , hope it's useful for you 
https://github.com/muhemuhe427/jsonlabel2yolo

**Comment by small-code-cat** [comment]
> 

Thank you very much！

**Comment by small-code-cat** [comment]
> I've wrote a script to transfer the label .json as .txt , and It is proved feasible for training , hope it's useful for you https://github.com/muhemuhe427/jsonlabel2yolo

model = torch.hub.load('ultralytics/yolov5', 'custom', 'yolov5m-seg.pt')
Hello, I use a picture to predict an image, and the results returned are shown in the picture. Can you tell me what these data mean? In addition, how to convert it to the json format in labelme
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/44658572/203917266-92174224-6410-42ed-897e-eb207aa2e1db.png">

**Comment by ryouchinsa** [comment]
You can convert the Labelme JSON files to the COCO format.
https://github.com/wkentaro/labelme/blob/main/examples/instance_segmentation/labelme2coco.py

You can convert the COCO JSON file to the YOLO segmentation format.
https://github.com/ultralytics/JSON2YOLO

[RectLabel](https://rectlabel.com) is an offline image annotation tool for object detection and segmentation.
Although this is not an open source program, with RectLabel you can import the Labelme JSON files and export to the YOLO segmentation format.

![yolo_polygon](https://user-images.githubusercontent.com/1954306/214472214-a9c2775d-247a-4bd1-87f6-9d90818ef8ff.jpg)

**Comment by dt1729** [comment]
I think this should be present in the main documentation of yolov5 for people who have created their own datasets without using the tools mentioned in the system.

**Comment by glenn-jocher** [comment]
@dt1729 you're right, the documentation is an important resource for users creating custom datasets. We'll take your feedback into consideration for future updates. If you have any other questions or need further assistance, feel free to ask.
